src/util/util.ts

```diff
@@ -91,7 +91,7 @@ export async function asyncMergeSort<T>(arr: T[], compareFn: SortCompareFn<T>):
 async function merge<T>(left: T[], right: T[], compareFn: SortCompareFn<T>) {
   const sorted: T[] = [];
   while (left.length > 0 && right.length > 0) {
-    if ((await compareFn(left[0], right[0])) < 0) {
+    if ((await compareFn(left[0], right[0])) <= 0) {
```

Comment:
Was this something we accidentally changed in our initial async work or did we have it wrong all along?

Reply:
In the original (non-async) code, we just used the array sort function built into javascript. But the built-in sort function requires a synchronous comparator function. When we switched to async, our comparator functions became async because they call `exec` -- so we could no longer use the built-in sort function.  @mgramigna looked for an open source async sort solution but could not find one; so he implemented an async-friendly merge sort function himself.

So I guess we "accidentally changed [it] in our initial async work" -- but it's not that we tweaked existing code, but rather that we had to re-implement something we used to get for free (and our re-implementation didn't behave quite the same as the previous built-in implementation).

Reply:
whoops!